napari/utils/migrations.py

```diff
@@ -211,3 +211,43 @@ def __init_subclass__(cls, **kwargs):
     _OldClass.__new__.__signature__ = prealloc_signature  # type: ignore [attr-defined]
 
     return _OldClass
+
+
+class _DeprecatingDict(dict[str, Any]):
```

Comment:
I really wanted to find an excuse to call this `SelfDeprecatingDict`, but that's not quite the right meaning here :p